src/CPDs/categorical_cpd.jl

```diff
@@ -37,7 +37,7 @@ function (cpd::CategoricalCPD)(a::Assignment=Assignment())
     if isempty(cpd.parents)
         return first(cpd.distributions)
     else
-        sub = [a[p] for p in cpd.parents]
+        sub = [a[potential] for potential in cpd.parents]
```

Comment:
the original `p` should stay. It is referring to the parents as `p`, not potentials.